trie/interface.go

```diff
@@ -18,9 +18,12 @@ type nodeData struct {
 	hexKey      []byte
 }
 
+type nodeWithHash struct {
```

Comment:
maybe add a comment describing the usage of this struct

Reply:
same for `nodeData`

Reply:
Added.